### PR TITLE
Update Makefile that runs generates to use the generate boilerplate instead of the go boilerplate

### PR DESCRIPTION
**Pull request by SataQiu**
**What this PR does / why we need it**:
Remove unused boilerplate.go.txt. 
`boilerplate.go.txt` is now in `hack/boilerplate/boilerplate.go.txt`

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by chuckha** [comment]
Yeah, I'd say hold off on this change but update the makefile that runs generates to use the generate boilerplate instead of the go boilerplate

File: Makefile [hunk 1]
```diff
@@ -91,7 +91,7 @@ vet:
 # Generate code
 .PHONY: generate
 generate: $(CONTROLLER_GEN)
-	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate.go.txt paths=./api/...
+	$(CONTROLLER_GEN) object:headerFile=./hack/boilerplate/boilerplate.go.txt paths=./api/...
```
Comment (chuckha) [inline Makefile]: this should use `boilerplate.generatego.txt` since generated files should not have the year in the copyright

**Comment by SataQiu** [comment]
Does this PR make sense? @chuckha @vincepri 
If not, I'll close it. Because it doesn't seem to be possible to remove `hack/boilerplate.go.txt` now.

**Comment by chuckha** [comment]
makes sense now that it's using generatego.txt!

/approve

**Pull request merged**